addons/binding/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/Connection.java

```diff
@@ -0,0 +1,115 @@
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Base64;
+
+import javax.net.ssl.HttpsURLConnection;
+
+/**
+ * @author Patrik Gfeller
+ *
+ */
+public class Connection {
+    private final String iCloudURL = "https://www.icloud.com";
+    private final String iCloudApiURL = "https://fmipmobile.icloud.com/fmipservice/device/";
+    private final String iCloudAPIRequestDataCommand = "/initClient";
+    private final String iCloudAPIPingDeviceCommand = "/playSound";
+
+    private final String dataRequest = "{\"clientContext\":{\"appName\":\"iCloud Find (Web)\",\"fmly\": true,\"appVersion\":\"2.0\",\"timezone\":\"US/Eastern\",\"inactiveTime\":2255,\"apiVersion\":\"3.0\",\"webStats\":\"0:15\"}}\n";
+
+    private final byte[] authorization;
+    private URL iCloudDataRequestURL;
+    private URL iCloudPingDeviceURL;
+
+    public Connection(String appleId, String password) throws MalformedURLException {
+        authorization = Base64.getEncoder().encode((appleId + ":" + password).getBytes());
+        iCloudDataRequestURL = new URL(iCloudApiURL + appleId + iCloudAPIRequestDataCommand);
+        iCloudPingDeviceURL = new URL(iCloudApiURL + appleId + iCloudAPIPingDeviceCommand);
+    }
+
+    public String requestDeviceStatusJSON() throws Exception {
+        HttpsURLConnection connection = postRequest(iCloudDataRequestURL, dataRequest);
+
+        String response = getResponse(connection);
+        return response;
+    }
+
+    /***
+     * Sends a ping request (find my phone).
+     *
+     * @throws Exception
+     */
+    public void pingPhone(String id) throws Exception {
+        String request = "{ \n \"device\": \"" + id + "\", \n \"subject\": \"Find My Device alert\" \n }";
+        getResponse(postRequest(iCloudPingDeviceURL, request));
+    }
+
+    private void setRequestProperties(HttpsURLConnection connection, String payload)
+            throws UnsupportedEncodingException {
+        connection.setRequestProperty("Authorization", this.getBasicAuthorization());
+        connection.setRequestProperty("User-Agent", "Find iPhone/1.3 MeKit (iPad: iPhone OS/4.2.1)");
+        connection.setRequestProperty("Origin", iCloudURL);
+        connection.setRequestProperty("Content-Type", "application/json");
+        connection.setRequestProperty("charset", "utf-8");
+        connection.setRequestProperty("Accept-language", "en-us");
+        connection.setRequestProperty("Connection", "keep-alive");
+        connection.setRequestProperty("X-Apple-Find-Api-Ver", "2.0");
+        connection.setRequestProperty("X-Apple-Authscheme", "UserIdGuest");
+        connection.setRequestProperty("X-Apple-Realm-Support", "1.0");
+        connection.setRequestProperty("X-Client-Name", "iPad");
+        connection.setRequestProperty("Content-Length", Integer.toString(payload.getBytes("UTF-8").length));
+    }
+
+    private HttpsURLConnection postRequest(URL url, String payload) throws Exception {
```

Comment:
Remove `throws Exception` check other places, if you call code which does this, please handle here or convert to a meaningful exception.